src/components/App/index.js

```diff
@@ -37,19 +37,25 @@ class App extends Component {
     if (id === "modalBg" || id === "closeModal") this.setState({ isModalVisible: false })
   }
 
+  setCurrentPage = e => {
+    const id = (Number(e.target.id));
+    this.props.history.push(`${id}`);
+  }
+
   render() {
-    const { error, loading, currentEpisodes } = this.props;
+    const { error, loading, episodes } = this.props;
     const { isModalVisible, activeEpisode, episodesPerPage } = this.state;
+    const currentPage = this.props.match.params.page;
+    const currentEpisodes = getCurrentEpisodes(currentPage, episodesPerPage, episodes);
 
     if (error) {
       return <p>{error.message}</p>;
     }
 
     return (
       <div className={styles.appWrapper}>
-        <Header/>
         { loading ? <Spinner /> : <EpisodesList episodes={currentEpisodes} day={today} openModal={this.openModal}/> }
-        <Pagination episodesPerPage={episodesPerPage}/>
+        <Pagination episodesPerPage={episodesPerPage} currentPage={currentPage} setCurrentPage={this.setCurrentPage}/>
```

Comment:
Do you need to pass `setCurrentPage` to the Pagination? Maybe you can connect this component directly to the store?

Reply:
First of all, please move `episodesPerPage` as const into Pagination or here in the App. It's const value so why it is in the state ? For now it's not going to change - if you like you can leave it as it is and in the next PR add select for the user to show 10/30/90/all episodes per page. Also please add pagination component also at the top, it's better UX.

`setCurrentPage` function can be move to the Pagination component?

Reply:
I agree in both cases. Corrected. :)